File: app/src/main/java/com/goranch/publicapis/di/PublicApisComponent.java
package com.goranch.publicapis.di;

import android.app.Application;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {PublicApisModule.class})
public interface PublicApisComponent {

    // Expose to subgraphs
    Application application();
}

File: app/src/main/java/com/goranch/publicapis/di/PublicApisComponent.kt
package com.goranch.publicapis.di

import android.app.Application
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(PublicApisModule::class))
interface PublicApisComponent {
    // Expose to subgraphs
    fun application(): Application
}